### ISSUE TITLE
No restriction on local network access

### ISSUE THREAD

**Issue by kosztyua** [LetheanMovement/lethean-vpn#29]
As discussed with ddvs1 Lucas is taking care of the issue and not disclosing yet. Just making a note here for the bounty :D

**Comment by limosek** [comment]
OK, so we should definitely discuss this :)
I was thinking about design and possibilities of haproxy. Even if haproxy is robust and powerfull tool, I would offload core security from there. it is somehow complicated because we are not working with simple destination IP but it needs to process headers to know next hop of request.
Haproxy is used just as authenticator and TLS encapsulator. It directs all requests to upstream proxy.
From this point of view, I suggest to rewrite documentation:

"VPN dispatcher should not be used as security filter. It only manages connections, encrypts data and receive payments.
We try to make it as secure and robust as possible but if you want to be absolutely sure that traffic from dispatcher server cannot flow to your infrastructure, use backend proxy ACLs and/or iptables."

In other words, create acls on upstream proxy to not allow local connections. 
For VPN mode, use iptables.
Securing squid is not so easy to do by sed or scripting, so easy-deploy-node will not do this by default. 
Please read squid documentation and create your acls there.
